src/py21cmfast/io/caching.py

```diff
@@ -39,25 +42,67 @@ class OutputCache:
     )
 
     _path_structures: ClassVar = {
-        "InitialConditions": "{user_cosmo:x}/{seed:d}/InitialConditions.h5",
-        "PerturbedField": "{user_cosmo:x}/{seed:d}/{zgrid:x}/{redshift:.4f}/PerturbedField.h5",
-        "other": "{user_cosmo:x}/{seed:d}/{zgrid:x}/{redshift:.4f}/{astro_flag:x}/{cls}.h5",
+        "InitialConditions": "{user_cosmo}/{seed}/InitialConditions.h5",
+        "PerturbedField": "{user_cosmo}/{seed}/{zgrid}/{redshift}/PerturbedField.h5",
+        "other": "{user_cosmo}/{seed}/{zgrid}/{redshift}/{astro_flag}/{cls}.h5",
     }
 
     @classmethod
     def _get_hashes(cls, inputs: InputParameters) -> dict[str, str]:
         """Return a dict of hashes for different components of the calculation."""
         # Python builtin hashes can be negative which looks weird in filenames
-        max_hash_value = 2**hash_info.width
         return {
-            "user_cosmo": hash((inputs.cosmo_params, inputs.user_params))
-            % max_hash_value,
-            "seed": inputs.random_seed % max_hash_value,
-            "zgrid": hash(inputs.node_redshifts) % max_hash_value,
-            "astro_flag": hash((inputs.astro_params, inputs.user_params))
-            % max_hash_value,
+            "user_cosmo": md5(
+                (repr(inputs.cosmo_params) + repr(inputs.user_params)).encode()
+            ).hexdigest(),
+            "seed": inputs.random_seed,
+            "zgrid": md5(repr(inputs.node_redshifts).encode()).hexdigest(),
+            "astro_flag": md5(
+                (repr(inputs.astro_params) + repr(inputs.user_params)).encode()
+            ).hexdigest(),
         }
 
+    @classmethod
+    def fill_path_template(
```

Comment:
I think we should make this `_fill_path_template` because it's not something that the users should be using.